pkg/helpers/newapp/app/imageref.go

```diff
@@ -44,6 +45,10 @@ type SecretAccessor interface {
 
 type imageRefGenerator struct{}
 
+var (
+	dnsNamePattern = regexp.MustCompile("[^a-zA-Z0-9-]")
```

Comment:
Upstream the DNS pattern for object names (DNS_SUBDOMAIN) is more nuanced:

https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/validation.go#L178
https://github.com/kubernetes/apimachinery/blob/71bf7ba06cd490392e05cd6828ae24a69cb0ccaa/pkg/util/validation/validation.go#L199

I would perhaps rename this to "dnsSubdomainAllowedCharacters"

```suggestion
	// Imagestream names must comply with k8s DNS1123 Subdomains
	dnsSubdomainAllowedCharacters = regexp.MustCompile("[^a-zA-Z0-9-]")
```